apps/namadillo/src/App/Swap/SwapSource.tsx

```diff
@@ -0,0 +1,94 @@
+import { AmountInput, Text } from "@namada/components";
+import { AvailableAmountFooter } from "App/Common/AvailableAmountFooter";
+import { SelectedAsset } from "App/Common/SelectedAsset";
+import { TokenAmountCard } from "App/Common/TokenAmountCard";
+import BigNumber from "bignumber.js";
+import clsx from "clsx";
+import { Asset } from "types";
+
+export type SwapSourceProps = {
+  walletAddress?: string;
```

Comment:
Now with these types they're all optional and become `string | undefined` but some are probably non-optional eh? :D

Reply:
I see your point but `string | undefined` is not a solution. It makes passing a prop to component mandatory, but it does not change the fact that the value can be undefined. So prop is not optional but the value inside prop still is. It only helps in scenarios in which someone forgets to pass a prop. Component still has to handle undefined. Components should not have so many potentially undefined props, because then you never know what you really get at which point. We should probably use more of this:
https://jotai.org/docs/guides/async#suspense